doc/source/templates/04_finetuning_llms_with_deepspeed/run_llama_ft.sh

```diff
@@ -54,7 +54,7 @@ fine_tune() {
         --test_path "${test_path}"  \
         --special_token_path "${token_path}" \
         --num-checkpoints-to-keep 1 \
-        --num-epochs 1 \
+        --num-epochs 10 \
```

Comment:
keep it at 1 by default

Reply:
Imo, 1 should be what happens if you do "--as-test". Right now "--as-test" skips evaluation and checkpointing. This can give false hopes. This PR makes it so that a a full epoch is tested with "--as-test" and a full training is run without. Isn't that the best thing to do? I've been using this script for a couple of days now and I find it very uncomforatble to run finetune_hf_llm.py manually every time, but rather always just want to either test or run a complete finetuning.

Reply:
Full epoch cannot be tested @ArturNiederfahrenhorst :) Let's keep as-test meaning to what it means. You should be using the python script directly if you are experimenting that much / or create your own shell script. This shell script is serving a simple entry point with default values for demo / test purposes. Not real training workloads.

Reply:
- When you say "keep as-test to what it means", what meaning are you refering to? Because the readme says "The flag `--as-test` is for demo / testing purposes as it runs through only one forward and backward pass of the model. The model loading, and remote checkpointing would still run. " so I'd expect that the meaning was that remote checkpointing would run. Also, I've used '-as-test' to test if I can run a full epoch, including checkpointing and everything so I could be quite sure that the workload was stable over multiple epochs. So the meaning of 'as-test' does not appear so crystal clear to me after all / the readme should probably say, that checkpointing is excluded and not tested.

- I was maybe a bit slow, but I'd say that I now fully understand what `run_llama_ft.sh` is there for 😄  In this case, I feel like there is something missing in this template:  An example that runs a full finetuning out of the box. Basically user can either run one forward/backward pass out of the box with `run_llama_ft.sh` or start reading / customizing to get to a full training. Is that intended? What if user just wants to run an end-to-end finetuning out of the box? That's how the other templates seem to work and I think it's a good idea!